tutorials/cloud-functions-async/index.md

```diff
@@ -0,0 +1,150 @@
+
+const (
+	Created JobState = iota
+	Running
+	Completed
+	Failed
+)
+
+type Job struct {
+	ID         string    `json:"id"`
+	CreateTime time.Time `json:"created-time"`
+	DoneTime   time.Time `json:"done-time" firestore:"DoneTime,omitempty"`
+	Done       bool
+	Result     string
+	State      JobState
+	Task       map[string]interface{} `json:"-"`
+	// can add a requester, source IP etc if needed
+}
+```
+
+When a job request is received, it is given an ID and the details of the work are included in a task field. This task is sent into the work queue as PubSub payload, with the Job ID as an [PubSub attribute](https://cloud.google.com/pubsub/docs/publisher#custom-attributes). When a worker picks up the task, it moves the job from `Created` -> `Running`. When the task is complete, it moves from `Running` to `Completed` (success) or `Failed` and writes any result back into the Firestore document.
+
+## Set Up
+
+1.  Create a project in the [Google Cloud Platform Console][console].
+1.  Enable billing for your project.
+1.  Use [Cloud Shell][shell] or Install the [Google Cloud SDK][sdk].
+
+[console]: https://console.cloud.google.com/
+[shell]: https://cloud.google.com/shell/
+[sdk]: https://cloud.google.com/sdk/
+
+Commands in this tutorial assume you are running from the tutorial folder:
+
+```
+git clone https://github.com/GoogleCloudPlatform/community.git
+cd community/tutorials/cloud-functions-async
+```
+
+### Create pubsub resources
+```
+gcloud pubsub topics create jobs
+gcloud pubsub subscriptions create worker-tasks --topic jobs
+```
+
+### Deploy Function
+
+`gcloud functions deploy Jobs --runtime go111 --trigger-http`
+
+## Create Jobs and check their state
+
+Once the function is deployed, we can use both POST and GET methods to interact with long running jobs.
+
+### Get URL
+
+Get the URL of the deployed function into an environment variable:
+
+```
+export URL=$(gcloud functions describe Jobs --format='value(httpsTrigger.url)')
+```
+
+### Create a Job
+
+Call the function based API to create a Job with a simple task. Our task has one field which is `worktime` which determines how many seconds the task takes to complete.
+
+```
+JOBID=$(curl -s --header "Content-Type: application/json"   --request POST   --data '{"worktime":40}'   $URL | jq -r '.id')
+```
+
+When the job is created, the function returns the ID of the job, which is put into an environment variable for reference later.
+
+### Check on Job status
+This command will poll the state of the job every 2 seconds with the [watch command](https://linux.die.net/man/1/watch).
+
+```
+watch -t "curl -s $URL/$JOBID | python -m json.tool"
+```
+
+### Perform the work
+
+Start a worker in another tab in either your local terminal or in Cloud Shell and run a basic worker that will complete after waiting the `worktime` value in the original 
+
+```
+cd worker
+go run main.go
+```
+
+## Cleanup and Next steps
+
+You can delete the deployed function and PubSub resources:
```

Comment:
I'd suggest either "Optionally, delete the deployed function and PubSub resources:" or "Delete the deployed function and PubSub resources:", depending on whether you want to leave them the option of keeping resources to use as a starting point for production code.